engine/src/test/java/io/camunda/zeebe/engine/state/migration/JobBackoffRestoreMigrationTest.java

```diff
@@ -135,6 +135,32 @@ public void shouldDoNothingIfFailedJobsAreTheSameAsBackoff() {
     assertThat(backoffColumnFamily.count()).isEqualTo(2);
   }
 
+  @Test
+  public void shouldNotRestoreJobWithoutRetries() {
+    // given
+    final MutableJobState jobState = processingState.getJobState();
+    final JobRecord record = createJobRecord(1000);
+    jobState.create(jobKey.getValue(), record);
+    jobState.fail(jobKey.getValue(), record);
+    backoffKey.wrapLong(record.getRecurringTime());
+    backoffColumnFamily.deleteExisting(backoffJobKey);
+    jobState.updateJobRetries(jobKey.getValue(), 0);
```

Comment:
:upside_down_face: If the `retires` are already set to `0` before the job is `failed` you won't need to remove it from the `backoffColumnFamily`. This would actually simulate the real-world behavior.

The current test still reproduces the bug in our code, but in a real-world scenario (if the job isn't removed from the `backoffColumnFamily`) it would still be valid, and the job would be made activatable as expected.

Reply:
Nice one 💪